src/core_ocean/analysis_members/mpas_ocn_okubo_weiss.F

```diff
@@ -0,0 +1,1767 @@
+
+   subroutine ocn_aggregate_eddy_stats(dmInfo, use_lat_lon_coords, min_cells, &!{{{
+                                       numCCs, maxCCsPerDomain, &
+                                       origCCId, sumVol, numCells, &
+                                       wsPosX, wsPosY, wsPosZ, &
+                                       wsVelX, wsVelY, wsVelZ, &
+                                       OW_cc_id, nVertLevels, nCellsSolve)
+
+
+      !-----------------------------------------------------------------
+      !
+      ! input variables
+      !
+      !-----------------------------------------------------------------
+
+      type (dm_info), intent(in) :: dminfo
+      logical, intent(in) :: use_lat_lon_coords
+      integer, intent(in) :: numCCs, maxCCsPerDomain, min_cells
+      real (kind=RKIND), dimension(:), intent(in) :: origCCId, sumVol, numCells, &
+                             wsPosX, wsPosY, wsPosZ, wsVelX, wsVelY, wsVelZ
+      integer, intent(in) :: nVertLevels, nCellsSolve
+
+      !-----------------------------------------------------------------
+      !
+      ! input/output variables
+      !
+      !-----------------------------------------------------------------
+
+      integer, dimension(:,:), intent(inout) :: OW_cc_id
+
+      !-----------------------------------------------------------------
+      !
+      ! local variables
+      !
+      !-----------------------------------------------------------------
+
+      real (kind=RKIND), dimension(9 * maxCCsPerDomain) :: stats
+      real (kind=RKIND), dimension(9 * maxCCsPerDomain * dmInfo % nprocs) :: &
+         globalStats, aggregated
+
+      integer i, j, k, ierr, curId, listIdx, totalCCs, cullId
+
+      ! Multiplex all eddy statistics into one array so it can be send with
+      ! one MPI communication call
+      call mpas_timer_start("all gather", .false., CCAggGatherT)
+
+      stats = 0
+      do i = 1, numCCs
+         stats(9*(i-1) + 1) = origCCId(i)
+         stats(9*(i-1) + 2) = sumVol(i)
+         stats(9*(i-1) + 3) = wsPosX(i)
+         stats(9*(i-1) + 4) = wsPosY(i)
+         stats(9*(i-1) + 5) = wsPosZ(i)
+         stats(9*(i-1) + 6) = wsVelX(i)
+         stats(9*(i-1) + 7) = wsVelY(i)
+         stats(9*(i-1) + 8) = wsVelZ(i)
+         stats(9*(i-1) + 9) = numCells(i)
+      end do
+
+      ! Send all local statistics
+      call MPI_Allgather(stats, 9*maxCCsPerDomain, MPI_REALKIND, &
+                      globalStats, 9*maxCCsPerDomain, MPI_REALKIND, &
+                      dmInfo % comm, ierr)
+      call mpas_timer_stop("all gather", CCAggGatherT)
+
+      ! Sort partial eddy stats according to eddy ID for efficient merging
+      call mpas_timer_start("sort/reduce", .false., CCAggSortT)
+
+      totalCCs = maxCCsPerDomain * dmInfo % nprocs
+      if (totalCCs > 0) then
+         call ocn_sort_eddy_stats(globalStats, totalCCs, 9)
+      end if
+
+      ! Aggregate the results
+      listIdx = 0
+      aggregated = 0
```

Comment:
@mark-petersen Can you specifically check this area to see if I updated the connected component array correctly? (The iteration over vert levels and cells, etc.)

I don't know if this is 100% correct or not (still learning MPAS).